app/models/chat_room.rb

```diff
@@ -1,33 +1,69 @@
+# == Schema Information
+#
+# Table name: chat_rooms
+#
+#  id            :bigint(8)        not null, primary key
+#  chatroom_type :integer
+#  identifier    :string
+#  title         :string
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  location_id   :bigint(8)
+#  system_id     :bigint(8)
+#
+# Indexes
+#
+#  index_chat_rooms_on_chatroom_type  (chatroom_type)
+#  index_chat_rooms_on_identifier     (identifier) UNIQUE
+#  index_chat_rooms_on_location_id    (location_id)
+#  index_chat_rooms_on_system_id      (system_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (location_id => locations.id)
+#  fk_rails_...  (system_id => systems.id)
+#
+
 class ChatRoom < ApplicationRecord
+  ensure_by :id, :identifier
+
+  ## -- RELATIONSHIPS
   belongs_to :location, optional: true
   has_and_belongs_to_many :users
   has_many :chat_messages, dependent: :destroy
   has_one :fleet, dependent: :destroy
   belongs_to :system, optional: true
 
+  ## -- VALIDATIONS
   validates :title, presence: true, length: { maximum: 20, too_long: I18n.t('validations.too_long_chat_room') }
+  validates :identifier, presence: true, uniqueness: true
 
+  ## -- ATTRIBUTES
   enum chatroom_type: [:global, :local, :custom, :corporation]
 
-  # Before Create -> Generate Identifier
-  before_create do
-    self.identifier = generate_random_identifier unless self.identifier && (self.identifier != "")
-  end
+  ## -- CALLBACKS
+  before_validation :set_identifier
 
-  # Update local players
+  ## — INSTANCE METHODS
   def update_local_players
-    ChatChannel.broadcast_to(self, method: 'update_players',
-                                   names: ApplicationController.helpers.map_and_sort(self.users.where("online > 0")),
-                                   fleet: self.fleet.present?,
-                                   color: (self.users.where("online > 0").sort_by { |obj| obj.full_name }.map { |p| p.active_spaceship.get_hp_color } if self.fleet.present?)
-                            )
+    if self.fleet.present?
+      color = self.users.is_online.in_name_order.map { |p| p.active_spaceship.get_hp_color }
+    end
+
+    ChatChannel.broadcast_to(self,
+      method: 'update_players',
+      names: ApplicationController.helpers.map_and_sort(self.users.is_online),
+      fleet: self.fleet.present?,
+      color: color
+    )
   end
 
-  private
+  def channel_id
+    "channel-#{self.identifier}"
+  end
 
-  # Generate random Identifier for ChatRoom
-  def generate_random_identifier
-    identifier = (0...8).map { (65 + rand(26)).chr }.join.upcase
-    ChatRoom.where(identifier: identifier).empty? ? identifier : generate_random_identifier
+  private
+  def set_identifier
+    self.identifier ||= SecureRandom.uuid
```

Comment:
What does `SecureRandom.uuid` generate? I need exactly 8 characters so its better to remember

Reply:
a uuid is longer -- but you _never_ have to check if it already exists or not as it's "universally unique"